.github/actions/runner-select/action.yml

```diff
@@ -104,3 +130,9 @@ runs:
         echo
         echo 'No self-hosted runners available!'
         fall_back_to_github_hosted
+
+    - name: Publish artifact marking this job as done
+      uses: actions/upload-artifact@v4
+      with:
```

Comment:
this should probably have an `if: ${{ always() }}`, so that this step also runs even if a previous step failed, or the job was cancelled.

Reply:
there wasn’t a timeout yet, but i could add one. do you mean something like rejecting if the workflow run started too long ago? in the meantime, added if always in 7a0a1d87b864de7c71a8d9b80bbd999daf047d9a

Reply:
Maybe even better idea: do not allow too old artifacts (max oldness being 5min or less). This way we can avoid another artifact. I would prefer if we could also remove the artifacts (I think this should happen from monitor), but I do not think we can relay on this to happen for security.

Reply:
Also one can always remove artifacts (not sure how many permissions is needed for this), so this could bypass the existing checks.

Reply:
> Maybe even better idea: do not allow too old artifacts (max oldness being 5min or less). This way we can avoid another artifact.

done in [ci-runners 1ee34d95455ea](https://github.com/servo/ci-runners/commit/1ee34d95455eabf73d33e1f4ad21bf10dd866012) and d86f99bb35ae91377ed1ca702648ede64ea6923b

> I would prefer if we could also remove the artifacts (I think this should happen from monitor), but I do not think we can relay on this to happen for security.

done in fbb472edff42d581dd7b85d9bfc0fbe58850ef8c, but i did it in the action, because we need to keep the amount of github API calls made by the monitors as low as possible to avoid rate limits. hopefully that’s ok, plus we don’t rely on it for security, and we set the lowest possible retention (7f4bbb8f1d469dfaa259654485c84e3e5d5cd4bc).